plugins/permission-node/api-report.md

```diff
@@ -119,4 +124,15 @@ export type PolicyDecision =
       result: AuthorizeResult.ALLOW | AuthorizeResult.DENY;
     }
   | ConditionalPolicyDecision;
+
+// @public
+export class ServerPermissionClient extends PermissionClient {
+  constructor(options: {
+    discoveryApi: DiscoveryApi;
+    configApi: Config;
+    serverTokenManager: TokenManager;
```

Comment:
As a general rule the name and type should match up for these kind of option objects, or the name should be shorter. Even `configApi` here should probably be just `Config` instead. On that topic I think we should probably use `PluginEndpointDiscovery` instead of `DiscoveryApi`, although it would be fine to have that named just `discovery`.

Regarding `serverTokenManager` here the type should be `ServerTokenManager` if we specifically want an instance of that class to be passed in here. If we're alright with any implementation of the `TokenManager` interface, `tokenManager` makes more sense. The fact that we currently only have two implementations of the `TokenManager` interface and we want people to pass in the `ServerTokenManager` right now doesn't matter at all. Integrators are completely free to whip up their own `TokenManager` implementation based on their own bespoke solutions, and the `ServerPermissionClient` should be fine with that.

Hoping that clears things up but happy to dive in more here if you want to :grin: